### PR TITLE
add export csv file for LF result

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/components/results/loadflow/load-flow-result.tsx [hunk 1]
```diff
@@ -256,14 +263,21 @@ export const LoadFlowResult: FunctionComponent<LoadflowResultProps> = ({
...
+                        id: 'LoadFlowResultsCurrentViolations',
+                    })}
+                    rows={rowsToShow}
+                    onRowDataUpdated={onRowDataUpdated}
+                    headerHeight={0}
                     onGridReady={onGridReady}
                     getRowStyle={getRowStyle}
                     overlayNoRowsTemplate={message}
+                    enableCellTextSelection={true}
+                    skipColumnHeaders={true}
```
Comment (AAJELLAL) [inline src/components/results/loadflow/load-flow-result.tsx:280]: Check with the PO if you need to export the header as well or not.

File: src/components/results/loadflow/load-flow-result.tsx [hunk 2]
```diff
@@ -309,14 +323,21 @@ export const LoadFlowResult: FunctionComponent<LoadflowResultProps> = ({
...
```
Comment (AAJELLAL) [inline src/components/results/loadflow/load-flow-result.tsx:340]: The same here.

File: src/components/results/loadflow/load-flow-result.tsx [hunk 3]
```diff
@@ -336,14 +357,21 @@ export const LoadFlowResult: FunctionComponent<LoadflowResultProps> = ({
...
+                        id: 'LoadFlowResultsStatus',
+                    })}
+                    rows={rowsToShow}
+                    onRowDataUpdated={onRowDataUpdated}
+                    headerHeight={0}
                     onGridReady={onGridReady}
                     getRowStyle={getRowStyle}
                     overlayNoRowsTemplate={message}
+                    enableCellTextSelection={true}
+                    skipColumnHeaders={true}
```
Comment (AAJELLAL) [inline src/components/results/loadflow/load-flow-result.tsx:374]: The same thing here.

**Review by AAJELLAL — CHANGES_REQUESTED** [review]
There is a regression in voltage init. The ag-grid table is no longer visible  when I select "INDICATORS" tab and the export crashes.
To fix it you need to change    **indicatorsColumnDefs={indicatorsColumnDefs}**  to                     **columns={indicatorsColumnDefs}**.
And also check with the PO if you need to keep he headers while exporting the result or not.

**Comment by ghelloumi** [comment]
- Headers heights are smaller.

**Review by ghelloumi — CHANGES_REQUESTED** [review]

File: src/components/utils/renderTable-ExportCsv.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    tableName: string;
+    rows: any[];
+    onRowDataUpdated: (event: RowDataUpdatedEvent<any, any>) => void;
+    headerHeight: number;
+    onGridReady: ((event: GridReadyEvent<any, any>) => void) | undefined;
+    getRowStyle: (params: RowClassParams<any, any>) => RowStyle | undefined;
+    enableCellTextSelection: boolean;
+    overlayNoRowsTemplate: string | undefined;
+    skipColumnHeaders: boolean;
+}
```
Comment (ghelloumi) [inline src/components/utils/renderTable-ExportCsv.tsx:40]: <any, any> are not necessary

```suggestion
interface RenderTableAndExportCsvProps {
    gridRef: Ref<any> | undefined;
    columns: any[];
    defaultColDef: ColDef<any>;
    tableName: string;
    rows: any[];
    onRowDataUpdated: (event: RowDataUpdatedEvent) => void;
    headerHeight: number;
    onGridReady: ((event: GridReadyEvent) => void) | undefined;
    getRowStyle: (params: RowClassParams) => RowStyle | undefined;
    enableCellTextSelection: boolean;
    overlayNoRowsTemplate: string | undefined;
    skipColumnHeaders: boolean;
}
```

File: src/components/utils/renderTable-ExportCsv.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    onGridReady,
+    getRowStyle,
+    enableCellTextSelection,
+    overlayNoRowsTemplate,
+    skipColumnHeaders = false,
+}) => {
+    return (
+        <Grid>
+            <Box sx={styles.csvExport}>
+                <Box style={{ flexGrow: 1 }}></Box>
```
Comment (ghelloumi) [inline src/components/utils/renderTable-ExportCsv.tsx:61]: idem, this Box is nt necessary, could you find a way to fix it please ?

**Review by ghelloumi — CHANGES_REQUESTED** [review]

**Review by ghelloumi — APPROVED** [review]

**Review by AAJELLAL — CHANGES_REQUESTED** [review]
The ag-grid table is no longer visible when I select the 3 first tabs but the  export works fine.

**Review by AAJELLAL — APPROVED** [review]

**Pull request merged**